e2e_tests/src/environment/nectar_instance.ts

```diff
@@ -0,0 +1,174 @@
+import { JsonMap, stringify } from "@iarna/toml";
+import { ChildProcess, spawn } from "child_process";
+import tempWrite from "temp-write";
+import waitForLogMessage from "./wait_for_log_message";
+import { Logger } from "log4js";
+import path from "path";
+import { crashListener } from "./crash_listener";
+import { openAsync, execAsync } from "./async_fs";
+import { NectarConfig } from "./nectar_config";
+import { sleep } from "../utils";
+import { parseFixed } from "@ethersproject/bignumber";
+
+export class NectarInstance {
+    private process: ChildProcess;
+
+    constructor(
+        private readonly cargoTargetDirectory: string,
+        private readonly logFile: string,
+        private readonly logger: Logger,
+        private readonly _config: NectarConfig
+    ) {}
+
+    public get config() {
+        return this._config;
+    }
+
+    public async deposit(): Promise<DepositAddresses> {
+        const bin = process.env.NECTAR_BIN
+            ? process.env.NECTAR_BIN
+            : path.join(this.cargoTargetDirectory, "debug", "nectar");
+
+        this.logger.info("Using binary", bin);
+
+        const configFile = await tempWrite(
+            stringify((this._config as unknown) as JsonMap),
+            "config.toml"
+        );
+
+        const { stdout } = await execAsync(
+            `${bin} --config=${configFile} --network=dev deposit`,
+            {
+                encoding: "utf-8",
+            }
+        );
+
+        return parseDepositOutput(stdout);
+    }
+
+    public async balance(): Promise<Balances> {
+        const bin = process.env.NECTAR_BIN
+            ? process.env.NECTAR_BIN
+            : path.join(this.cargoTargetDirectory, "debug", "nectar");
+
+        this.logger.info("Using binary", bin);
+
+        const configFile = await tempWrite(
+            stringify((this._config as unknown) as JsonMap),
+            "config.toml"
+        );
+
+        const { stdout } = await execAsync(
+            `${bin} --config=${configFile} --network=dev balance`,
+            {
+                encoding: "utf-8",
+            }
+        );
+
+        return parseBalanceOutput(stdout);
+    }
+
+    /*
+     * Executes nectar's `trade` command.
+     *
+     * Returns the PeerId under which this nectar instance participates in the COMIT network.
+     */
+    public async trade(): Promise<string> {
+        const bin = process.env.NECTAR_BIN
```

Comment:
You are doing that few times, you may want to extract to a function

Reply:
Actually, I don't think we are using this functionality at all. Might as well remove it :D

Reply:
Actually, I can't find it in the CI config. Are we no longer testing against the release build?

Reply:
Testing against the release build is done as part of the pre-release CI as far as I remember.

Reply:
That is what I was thinking but I couldn't find that anywhere ...